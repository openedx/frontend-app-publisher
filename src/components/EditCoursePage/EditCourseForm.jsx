import React from 'react';
import { Field, reduxForm } from 'redux-form';
import PropTypes from 'prop-types';

import RenderInputTextField from '../RenderInputTextField';
import RenderSelectField from '../RenderSelectField';
import ImageUpload from '../../components/ImageUpload';
import RichEditor from '../../components/RichEditor';
import { AUDIT_TRACK, VERIFIED_TRACK, PROFESSIONAL_TRACK } from '../../data/constants';


class EditCourseForm extends React.Component {
  getEnrollmentTrackOptions() {
    return [
      { label: VERIFIED_TRACK.name, value: VERIFIED_TRACK.key },
      { label: AUDIT_TRACK.name, value: AUDIT_TRACK.key },
      { label: PROFESSIONAL_TRACK.name, value: PROFESSIONAL_TRACK.key },
    ];
  }

  getOptions() {
    const { courseOptions } = this.props;

    if (!courseOptions) {
      return [];
    }

    const { data } = courseOptions;

    if (!data || !data.actions) {
      return [];
    }

    return data.actions.PUT;
  }

  parseOptions(inChoices) {
    return inChoices.map(choice =>
      ({ label: choice.display_name, value: choice.value }));
  }

  render() {
    const {
      handleSubmit,
      number,
      entitlement,
      submitting,
    } = this.props;

    const options = this.getOptions();
    const levelTypeOptions = options && this.parseOptions(options.level_type.choices);
    const subjectOptions = options && this.parseOptions(options.subjects.child.choices);

    levelTypeOptions.unshift({ label: '--', value: '' });
    subjectOptions.unshift({ label: '--', value: '' });

    return (
      <div className="edit-course-form row">
        <div className="col">
          <form onSubmit={handleSubmit}>
            <Field
              name="title"
              component={RenderInputTextField}
              type="text"
              label={<strong>Title: *</strong>}
              required
            />
            <div className="row">
              <div id="number" className="col-12">
                <strong>Number: *</strong>
              </div>
              <div className="col-12">{number}</div>
            </div>
            <Field
              name="short_description"
              component={RichEditor}
              label={<strong>Short Description: *</strong>}
              maxChars={500}
              id="sdesc"
            />
            <Field
              name="full_description"
              component={RichEditor}
              label={<strong>Long Description: *</strong>}
              maxChars={2500}
              id="ldesc"
            />
            <Field
              name="outcome"
              component={RichEditor}
              label={<strong>What you will learn: *</strong>}
              maxChars={2500}
              id="outcome"
            />
            <Field
              name="subjectPrimary"
              component={RenderSelectField}
              label={<strong>Primary Subject: *</strong>}
              options={subjectOptions}
              required
            />
            <Field
              name="subjectSecondary"
              component={RenderSelectField}
              label={<strong>Secondary Subject:</strong>}
              options={subjectOptions}
            />
            <Field
              name="subjectTertiary"
              component={RenderSelectField}
              label={<strong>Tertiary Subject:</strong>}
              options={subjectOptions}
            />
            <Field
              name="imageSrc"
              component={ImageUpload}
              label={<strong>Image: *</strong>}
              id="image"
            />
            <Field
              name="prerequisites_raw"
              component={RichEditor}
              label={<strong>Prerequisites:</strong>}
              maxChars={1000}
              id="prereq"
            />
            <Field
              name="level_type"
              component={RenderSelectField}
              label={<strong>Course Level: *</strong>}
              options={levelTypeOptions}
              required
            />
            <Field
              name="learner_testimonials"
              component={RichEditor}
              label={<strong>Learner Testimonials:</strong>}
              maxChars={500}
              id="learner-testimonials"
            />
            <Field
              name="faq"
              component={RichEditor}
              label={<strong>Frequently Asked Questions:</strong>}
              maxChars={2500}
              id="faq"
            />
            <Field
              name="additional_information"
              component={RichEditor}
              label={<strong>Additional Information:</strong>}
              maxChars={2500}
              id="additional-information"
            />
            <Field
              name="syllabus_raw"
              component={RichEditor}
              label={<strong>Syllabus:</strong>}
              maxChars={500}
              id="syllabus"
            />
            <Field
              name="videoSrc"
              component={RenderInputTextField}
              type="text"
              label={<strong>About Video Link: *</strong>}
              required
            />
            {entitlement && (
              <React.Fragment>
                <Field
                  name="mode"
                  component={RenderSelectField}
                  label={<strong>Enrollment Track:</strong>}
                  options={this.getEnrollmentTrackOptions()}
                />
                <Field
                  name="price"
                  component={RenderInputTextField}
                  type="number"
                  label={<strong>Price: *</strong>}
                />
              </React.Fragment>
            )}
            <div className="row justify-content-end">
              <button type="submit" className="btn btn-outline-primary form-submit-btn" disabled={submitting}>
                Save Course
              </button>
            </div>
          </form>
        </div>
      </div>
    );
  }
}

EditCourseForm.defaultProps = {
  entitlement: false,
};

EditCourseForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  number: PropTypes.string.isRequired,
  entitlement: PropTypes.bool,
  courseOptions: PropTypes.shape({
    data: PropTypes.shape(),
    error: PropTypes.string,
    isFetching: PropTypes.bool,
  }).isRequired,
  submitting: PropTypes.bool.isRequired,
};

export default reduxForm({ form: 'edit-course-form' })(EditCourseForm);
