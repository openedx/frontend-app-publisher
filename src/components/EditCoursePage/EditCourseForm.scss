.edit-course-form {
  .course-image {
    .uploaded-image {
      max: {
        // These match the 'small' size in course-discovery's model definition for the image field
        width: 378px;
        height: 225px;
      }
    }
  }

  .new-run-button {
    border: 1px dashed silver;
  }

  .course-skill {
    border-radius: 15px;
    padding: 10px;
    font-size: 0.9rem;
    margin: {
      right: 5px;
      top: 6px;
    }
  }

  .react-select-async__menu {
    z-index: 5;
  }

  .react-select-async__control {
    margin-top: 8px;
  }

}
