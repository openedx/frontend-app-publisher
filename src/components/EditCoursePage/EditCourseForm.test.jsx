import React from 'react';
import { shallow } from 'enzyme';

import EditCourseForm from './EditCourseForm';


describe('EditCourseForm', () => {
  const courseOptions = {
    data: {
      actions: {
        PUT: {
          level_type: {
            choices: [
              { display_name: 'Beginner', value: 'beginner' },
              { display_name: 'Intermediate', value: 'intermediate' },
              { display_name: 'Advanced', value: 'advanced' },
            ],
          },
          subjects: {
            child: {
              choices: [
                { display_name: 'Business', value: 'business' },
                { display_name: 'Chemistry', value: 'chemistry' },
                { display_name: 'English', value: 'english' },
                { display_name: 'Security', value: 'security' },
              ],
            },
          },
        },
      },
    },
  };
  const initialValuesFull = {
    title: 'Test Title',
    short_description: 'short desc',
    full_description: 'long desc',
    outcome: 'learning outcomes',
    subjectPrimary: 'business',
    subjectSecondary: 'english',
    subjectTertiary: 'security',
    imageSrc: 'http://image.src.small',
    prerequisites_raw: 'prereq',
    level_type: 'advanced',
    learner_testimonials: 'learner testimonials',
    faq: 'frequently asked questions',
    additional_information: 'additional info',
    syllabus_raw: 'syllabus',
    videoSrc: 'https://www.video.src/watch?v=fdsafd',
    mode: 'verified',
    price: '77',
  };

  it('renders html correctly with minimal data', () => {
    const component = shallow(<EditCourseForm
      handleSubmit={() => null}
      initialValues={{
        title: 'Test Title',
      }}
      number="Test101x"
      courseOptions={courseOptions}
    />);
    expect(component).toMatchSnapshot();
  });

  it('renders html correctly with all data present and verified entitlement', () => {
    const component = shallow(<EditCourseForm
      handleSubmit={() => null}
      initialValues={initialValuesFull}
      number="Test102x"
      courseOptions={courseOptions}
      entitlement
    />);
    expect(component).toMatchSnapshot();
  });

  it('renders html correctly while submitting', () => {
    const component = shallow(<EditCourseForm
      handleSubmit={() => null}
      initialValues={initialValuesFull}
      number="Test103x"
      courseOptions={courseOptions}
      entitlement
      submitting
    />);
    expect(component).toMatchSnapshot();
  });
});
