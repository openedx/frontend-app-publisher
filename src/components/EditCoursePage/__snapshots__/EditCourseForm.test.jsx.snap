// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditCourseForm renders html correctly while submitting 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReduxForm
    courseOptions={
      Object {
        "data": Object {
          "actions": Object {
            "PUT": Object {
              "level_type": Object {
                "choices": Array [
                  Object {
                    "display_name": "Beginner",
                    "value": "beginner",
                  },
                  Object {
                    "display_name": "Intermediate",
                    "value": "intermediate",
                  },
                  Object {
                    "display_name": "Advanced",
                    "value": "advanced",
                  },
                ],
              },
              "subjects": Object {
                "child": Object {
                  "choices": Array [
                    Object {
                      "display_name": "Business",
                      "value": "business",
                    },
                    Object {
                      "display_name": "Chemistry",
                      "value": "chemistry",
                    },
                    Object {
                      "display_name": "English",
                      "value": "english",
                    },
                    Object {
                      "display_name": "Security",
                      "value": "security",
                    },
                  ],
                },
              },
            },
          },
        },
      }
    }
    entitlement={true}
    handleSubmit={[Function]}
    initialValues={
      Object {
        "additional_information": "additional info",
        "faq": "frequently asked questions",
        "full_description": "long desc",
        "imageSrc": "http://image.src.small",
        "learner_testimonials": "learner testimonials",
        "level_type": "advanced",
        "mode": "verified",
        "outcome": "learning outcomes",
        "prerequisites_raw": "prereq",
        "price": "77",
        "short_description": "short desc",
        "subjectPrimary": "business",
        "subjectSecondary": "english",
        "subjectTertiary": "security",
        "syllabus_raw": "syllabus",
        "title": "Test Title",
        "videoSrc": "https://www.video.src/watch?v=fdsafd",
      }
    }
    number="Test103x"
    submitting={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "courseOptions": Object {
        "data": Object {
          "actions": Object {
            "PUT": Object {
              "level_type": Object {
                "choices": Array [
                  Object {
                    "display_name": "Beginner",
                    "value": "beginner",
                  },
                  Object {
                    "display_name": "Intermediate",
                    "value": "intermediate",
                  },
                  Object {
                    "display_name": "Advanced",
                    "value": "advanced",
                  },
                ],
              },
              "subjects": Object {
                "child": Object {
                  "choices": Array [
                    Object {
                      "display_name": "Business",
                      "value": "business",
                    },
                    Object {
                      "display_name": "Chemistry",
                      "value": "chemistry",
                    },
                    Object {
                      "display_name": "English",
                      "value": "english",
                    },
                    Object {
                      "display_name": "Security",
                      "value": "security",
                    },
                  ],
                },
              },
            },
          },
        },
      },
      "destroyOnUnmount": true,
      "enableReinitialize": false,
      "entitlement": true,
      "forceUnregisterOnUnmount": false,
      "form": "edit-course-form",
      "getFormState": [Function],
      "handleSubmit": [Function],
      "initialValues": Object {
        "additional_information": "additional info",
        "faq": "frequently asked questions",
        "full_description": "long desc",
        "imageSrc": "http://image.src.small",
        "learner_testimonials": "learner testimonials",
        "level_type": "advanced",
        "mode": "verified",
        "outcome": "learning outcomes",
        "prerequisites_raw": "prereq",
        "price": "77",
        "short_description": "short desc",
        "subjectPrimary": "business",
        "subjectSecondary": "english",
        "subjectTertiary": "security",
        "syllabus_raw": "syllabus",
        "title": "Test Title",
        "videoSrc": "https://www.video.src/watch?v=fdsafd",
      },
      "keepDirtyOnReinitialize": false,
      "number": "Test103x",
      "persistentSubmitErrors": false,
      "pure": true,
      "shouldAsyncValidate": [Function],
      "shouldError": [Function],
      "shouldValidate": [Function],
      "shouldWarn": [Function],
      "submitting": true,
      "touchOnBlur": true,
      "touchOnChange": false,
      "updateUnregisteredFields": false,
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "courseOptions": Object {
          "data": Object {
            "actions": Object {
              "PUT": Object {
                "level_type": Object {
                  "choices": Array [
                    Object {
                      "display_name": "Beginner",
                      "value": "beginner",
                    },
                    Object {
                      "display_name": "Intermediate",
                      "value": "intermediate",
                    },
                    Object {
                      "display_name": "Advanced",
                      "value": "advanced",
                    },
                  ],
                },
                "subjects": Object {
                  "child": Object {
                    "choices": Array [
                      Object {
                        "display_name": "Business",
                        "value": "business",
                      },
                      Object {
                        "display_name": "Chemistry",
                        "value": "chemistry",
                      },
                      Object {
                        "display_name": "English",
                        "value": "english",
                      },
                      Object {
                        "display_name": "Security",
                        "value": "security",
                      },
                    ],
                  },
                },
              },
            },
          },
        },
        "destroyOnUnmount": true,
        "enableReinitialize": false,
        "entitlement": true,
        "forceUnregisterOnUnmount": false,
        "form": "edit-course-form",
        "getFormState": [Function],
        "handleSubmit": [Function],
        "initialValues": Object {
          "additional_information": "additional info",
          "faq": "frequently asked questions",
          "full_description": "long desc",
          "imageSrc": "http://image.src.small",
          "learner_testimonials": "learner testimonials",
          "level_type": "advanced",
          "mode": "verified",
          "outcome": "learning outcomes",
          "prerequisites_raw": "prereq",
          "price": "77",
          "short_description": "short desc",
          "subjectPrimary": "business",
          "subjectSecondary": "english",
          "subjectTertiary": "security",
          "syllabus_raw": "syllabus",
          "title": "Test Title",
          "videoSrc": "https://www.video.src/watch?v=fdsafd",
        },
        "keepDirtyOnReinitialize": false,
        "number": "Test103x",
        "persistentSubmitErrors": false,
        "pure": true,
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "submitting": true,
        "touchOnBlur": true,
        "touchOnChange": false,
        "updateUnregisteredFields": false,
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`EditCourseForm renders html correctly with all data present and verified entitlement 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReduxForm
    courseOptions={
      Object {
        "data": Object {
          "actions": Object {
            "PUT": Object {
              "level_type": Object {
                "choices": Array [
                  Object {
                    "display_name": "Beginner",
                    "value": "beginner",
                  },
                  Object {
                    "display_name": "Intermediate",
                    "value": "intermediate",
                  },
                  Object {
                    "display_name": "Advanced",
                    "value": "advanced",
                  },
                ],
              },
              "subjects": Object {
                "child": Object {
                  "choices": Array [
                    Object {
                      "display_name": "Business",
                      "value": "business",
                    },
                    Object {
                      "display_name": "Chemistry",
                      "value": "chemistry",
                    },
                    Object {
                      "display_name": "English",
                      "value": "english",
                    },
                    Object {
                      "display_name": "Security",
                      "value": "security",
                    },
                  ],
                },
              },
            },
          },
        },
      }
    }
    entitlement={true}
    handleSubmit={[Function]}
    initialValues={
      Object {
        "additional_information": "additional info",
        "faq": "frequently asked questions",
        "full_description": "long desc",
        "imageSrc": "http://image.src.small",
        "learner_testimonials": "learner testimonials",
        "level_type": "advanced",
        "mode": "verified",
        "outcome": "learning outcomes",
        "prerequisites_raw": "prereq",
        "price": "77",
        "short_description": "short desc",
        "subjectPrimary": "business",
        "subjectSecondary": "english",
        "subjectTertiary": "security",
        "syllabus_raw": "syllabus",
        "title": "Test Title",
        "videoSrc": "https://www.video.src/watch?v=fdsafd",
      }
    }
    number="Test102x"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "courseOptions": Object {
        "data": Object {
          "actions": Object {
            "PUT": Object {
              "level_type": Object {
                "choices": Array [
                  Object {
                    "display_name": "Beginner",
                    "value": "beginner",
                  },
                  Object {
                    "display_name": "Intermediate",
                    "value": "intermediate",
                  },
                  Object {
                    "display_name": "Advanced",
                    "value": "advanced",
                  },
                ],
              },
              "subjects": Object {
                "child": Object {
                  "choices": Array [
                    Object {
                      "display_name": "Business",
                      "value": "business",
                    },
                    Object {
                      "display_name": "Chemistry",
                      "value": "chemistry",
                    },
                    Object {
                      "display_name": "English",
                      "value": "english",
                    },
                    Object {
                      "display_name": "Security",
                      "value": "security",
                    },
                  ],
                },
              },
            },
          },
        },
      },
      "destroyOnUnmount": true,
      "enableReinitialize": false,
      "entitlement": true,
      "forceUnregisterOnUnmount": false,
      "form": "edit-course-form",
      "getFormState": [Function],
      "handleSubmit": [Function],
      "initialValues": Object {
        "additional_information": "additional info",
        "faq": "frequently asked questions",
        "full_description": "long desc",
        "imageSrc": "http://image.src.small",
        "learner_testimonials": "learner testimonials",
        "level_type": "advanced",
        "mode": "verified",
        "outcome": "learning outcomes",
        "prerequisites_raw": "prereq",
        "price": "77",
        "short_description": "short desc",
        "subjectPrimary": "business",
        "subjectSecondary": "english",
        "subjectTertiary": "security",
        "syllabus_raw": "syllabus",
        "title": "Test Title",
        "videoSrc": "https://www.video.src/watch?v=fdsafd",
      },
      "keepDirtyOnReinitialize": false,
      "number": "Test102x",
      "persistentSubmitErrors": false,
      "pure": true,
      "shouldAsyncValidate": [Function],
      "shouldError": [Function],
      "shouldValidate": [Function],
      "shouldWarn": [Function],
      "touchOnBlur": true,
      "touchOnChange": false,
      "updateUnregisteredFields": false,
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "courseOptions": Object {
          "data": Object {
            "actions": Object {
              "PUT": Object {
                "level_type": Object {
                  "choices": Array [
                    Object {
                      "display_name": "Beginner",
                      "value": "beginner",
                    },
                    Object {
                      "display_name": "Intermediate",
                      "value": "intermediate",
                    },
                    Object {
                      "display_name": "Advanced",
                      "value": "advanced",
                    },
                  ],
                },
                "subjects": Object {
                  "child": Object {
                    "choices": Array [
                      Object {
                        "display_name": "Business",
                        "value": "business",
                      },
                      Object {
                        "display_name": "Chemistry",
                        "value": "chemistry",
                      },
                      Object {
                        "display_name": "English",
                        "value": "english",
                      },
                      Object {
                        "display_name": "Security",
                        "value": "security",
                      },
                    ],
                  },
                },
              },
            },
          },
        },
        "destroyOnUnmount": true,
        "enableReinitialize": false,
        "entitlement": true,
        "forceUnregisterOnUnmount": false,
        "form": "edit-course-form",
        "getFormState": [Function],
        "handleSubmit": [Function],
        "initialValues": Object {
          "additional_information": "additional info",
          "faq": "frequently asked questions",
          "full_description": "long desc",
          "imageSrc": "http://image.src.small",
          "learner_testimonials": "learner testimonials",
          "level_type": "advanced",
          "mode": "verified",
          "outcome": "learning outcomes",
          "prerequisites_raw": "prereq",
          "price": "77",
          "short_description": "short desc",
          "subjectPrimary": "business",
          "subjectSecondary": "english",
          "subjectTertiary": "security",
          "syllabus_raw": "syllabus",
          "title": "Test Title",
          "videoSrc": "https://www.video.src/watch?v=fdsafd",
        },
        "keepDirtyOnReinitialize": false,
        "number": "Test102x",
        "persistentSubmitErrors": false,
        "pure": true,
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "touchOnBlur": true,
        "touchOnChange": false,
        "updateUnregisteredFields": false,
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`EditCourseForm renders html correctly with minimal data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReduxForm
    courseOptions={
      Object {
        "data": Object {
          "actions": Object {
            "PUT": Object {
              "level_type": Object {
                "choices": Array [
                  Object {
                    "display_name": "Beginner",
                    "value": "beginner",
                  },
                  Object {
                    "display_name": "Intermediate",
                    "value": "intermediate",
                  },
                  Object {
                    "display_name": "Advanced",
                    "value": "advanced",
                  },
                ],
              },
              "subjects": Object {
                "child": Object {
                  "choices": Array [
                    Object {
                      "display_name": "Business",
                      "value": "business",
                    },
                    Object {
                      "display_name": "Chemistry",
                      "value": "chemistry",
                    },
                    Object {
                      "display_name": "English",
                      "value": "english",
                    },
                    Object {
                      "display_name": "Security",
                      "value": "security",
                    },
                  ],
                },
              },
            },
          },
        },
      }
    }
    handleSubmit={[Function]}
    initialValues={
      Object {
        "title": "Test Title",
      }
    }
    number="Test101x"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "courseOptions": Object {
        "data": Object {
          "actions": Object {
            "PUT": Object {
              "level_type": Object {
                "choices": Array [
                  Object {
                    "display_name": "Beginner",
                    "value": "beginner",
                  },
                  Object {
                    "display_name": "Intermediate",
                    "value": "intermediate",
                  },
                  Object {
                    "display_name": "Advanced",
                    "value": "advanced",
                  },
                ],
              },
              "subjects": Object {
                "child": Object {
                  "choices": Array [
                    Object {
                      "display_name": "Business",
                      "value": "business",
                    },
                    Object {
                      "display_name": "Chemistry",
                      "value": "chemistry",
                    },
                    Object {
                      "display_name": "English",
                      "value": "english",
                    },
                    Object {
                      "display_name": "Security",
                      "value": "security",
                    },
                  ],
                },
              },
            },
          },
        },
      },
      "destroyOnUnmount": true,
      "enableReinitialize": false,
      "forceUnregisterOnUnmount": false,
      "form": "edit-course-form",
      "getFormState": [Function],
      "handleSubmit": [Function],
      "initialValues": Object {
        "title": "Test Title",
      },
      "keepDirtyOnReinitialize": false,
      "number": "Test101x",
      "persistentSubmitErrors": false,
      "pure": true,
      "shouldAsyncValidate": [Function],
      "shouldError": [Function],
      "shouldValidate": [Function],
      "shouldWarn": [Function],
      "touchOnBlur": true,
      "touchOnChange": false,
      "updateUnregisteredFields": false,
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "courseOptions": Object {
          "data": Object {
            "actions": Object {
              "PUT": Object {
                "level_type": Object {
                  "choices": Array [
                    Object {
                      "display_name": "Beginner",
                      "value": "beginner",
                    },
                    Object {
                      "display_name": "Intermediate",
                      "value": "intermediate",
                    },
                    Object {
                      "display_name": "Advanced",
                      "value": "advanced",
                    },
                  ],
                },
                "subjects": Object {
                  "child": Object {
                    "choices": Array [
                      Object {
                        "display_name": "Business",
                        "value": "business",
                      },
                      Object {
                        "display_name": "Chemistry",
                        "value": "chemistry",
                      },
                      Object {
                        "display_name": "English",
                        "value": "english",
                      },
                      Object {
                        "display_name": "Security",
                        "value": "security",
                      },
                    ],
                  },
                },
              },
            },
          },
        },
        "destroyOnUnmount": true,
        "enableReinitialize": false,
        "forceUnregisterOnUnmount": false,
        "form": "edit-course-form",
        "getFormState": [Function],
        "handleSubmit": [Function],
        "initialValues": Object {
          "title": "Test Title",
        },
        "keepDirtyOnReinitialize": false,
        "number": "Test101x",
        "persistentSubmitErrors": false,
        "pure": true,
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "touchOnBlur": true,
        "touchOnChange": false,
        "updateUnregisteredFields": false,
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
